src/client/components/network-import/excel-import-wizard.js

```diff
@@ -210,20 +211,40 @@ class ExcelImportSubWizard extends React.Component {
 
   async handleFinish() {
     const { networkName, edgeData, nodeData } = this.state;
+
+    const data = { name: networkName };
     const elements = this.createCyElements(edgeData, nodeData);
 
-    if (elements && elements.length > 0) {
-      try {
-        this.controller.setNetwork(elements, { name: networkName });
-      } catch (e) {
-        console.log(e); // TODO Show error to user
-      }
+    // Create another Cytoscape instance, just so we can apply a layout
+    const cy = new Cytoscape();
+    cy.data(data);
+    cy.add(elements);
+
+    // Apply grid layout (but first, calculate an adequate bounding box where nodes don't overlap)
+    const nc = cy.nodes().length;
+    const size = 50 * Math.round(Math.sqrt(nc));
+    const w = 1.25 * size;
+    const h = 0.75 * size;
+    cy.layout({ name: 'grid', boundingBox: { x1: 0, y1: 0, w, h } }).run();
```

Comment:
You can alternatively have the layout automatically avoid overlap: https://js.cytoscape.org/#layouts/grid

You need the style enabled on the headless instance to do that.  Otherwise, the layout can't know how big the nodes are.